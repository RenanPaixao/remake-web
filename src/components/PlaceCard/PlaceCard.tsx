import { Button, Card, CardFooter, CardHeader, CardProps, Heading, Text } from '@chakra-ui/react'

interface IProps extends CardProps{
  title: string
  distance: string
  address: string
  onClick: () => void
}

export const PlaceCard = (props: IProps) => {
  const { flexBasis = '40rem', onClick, title, distance, address, ...rest } = props
  return <Card {...rest} variant={'elevated'} size={'md'} px={2} flexBasis={flexBasis} boxShadow={'rgba(0, 0, 0, 0.07) 0px 1px 2px, rgba(0, 0, 0, 0.07) 0px 2px 4px, rgba(0, 0, 0, 0.07) 0px 4px 8px, rgba(0, 0, 0, 0.07) 0px 8px 16px, rgba(0, 0, 0, 0.07) 0px 16px 32px, rgba(0, 0, 0, 0.07) 0px 32px 64px'}>
    <CardHeader display={'flex'} justifyContent={'space-between'} alignItems={'center'}>
      <Heading as={'h3'} size={'md'} textAlign={'left'}>
        {title}
      </Heading>

      <Text ml={2} as={'h5'} size={'sm'} textAlign={'left'}>
        {distance}
      </Text>
    </CardHeader>

    <CardFooter alignItems={'center'}>
      <Text display={'box'} pr={'1rem'} textAlign={'left'}>{address}</Text>
      <Button onClick={onClick} colorScheme={'blue'} ml={'auto'}>Go!</Button>
    </CardFooter>
  </Card>
}
