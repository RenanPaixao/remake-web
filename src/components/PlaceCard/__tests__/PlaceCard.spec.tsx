import { expect } from 'vitest'
import { customRender, user } from '../../../../tests/test-utils.tsx'
import { PlaceCard } from '../PlaceCard.tsx'

describe('PlaceCard', () => {
  it('should render', async () => {
    const { container } = customRender(
      <PlaceCard
        title={'Place Title'}
        address={'Place Address'}
        distance={'60 KM'}
        onClick={() => vi.fn()}
      />)

    expect(container).toMatchSnapshot()
  })

  it('should call action', async () => {
    const onClick = vi.fn()
    const { getByRole } = customRender(
      <PlaceCard
        title={'Place Title'}
        address={'Place Address'}
        distance={'60 KM'}
        onClick={onClick}
      />)

    const button = getByRole('button')
    await user.click(button)

    expect(onClick).toHaveBeenCalledOnce()
  })
})
