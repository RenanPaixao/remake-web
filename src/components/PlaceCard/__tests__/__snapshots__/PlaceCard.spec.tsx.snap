// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlaceCard > should render 1`] = `
<div>
  <div
    class="chakra-card css-1101n89"
  >
    <div
      class="chakra-card__header css-1fj4epu"
    >
      <h3
        class="chakra-heading css-1q6o3tu"
      >
        Place Title
      </h3>
      <h5
        class="chakra-text css-1vuos2j"
      >
        60 KM
      </h5>
    </div>
    <div
      class="chakra-card__footer css-1r7igq6"
    >
      <p
        class="chakra-text css-cg4qa7"
      >
        Place Address
      </p>
      <button
        class="chakra-button css-1noe2cj"
        type="button"
      >
        Go!
      </button>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
